sdk/identity/azure-identity/azure/identity/_credentials/authorization_code.py

```diff
@@ -86,7 +86,12 @@ def get_token(
           ``response`` attribute.
         """
         # pylint:disable=useless-super-delegation
-        return super(AuthorizationCodeCredential, self).get_token(*scopes, claims=claims, tenant_id=tenant_id, **kwargs)
+        client_secret = (
+            kwargs.pop("client_secret", None) or self._client_secret
+        )  # If secret not provided in method, use init secret
```

Comment:
We already have self._client_secret = kwargs.pop("client_secret", None) in the ctor, I don't think we need to expose `client_secret` in `get_token()` method.

Reply:
What's in my mind is we can test `client_secret` is correctly added into the request. Your ideas?

Reply:
I was able to patch that in and get the flow to pass the tests. That is what i was thinking but i wanted to confirm that there was not another valid path that did not involve the `client_secret`.

Reply:
If there is one that we did not know, we can add another test to cover that one. :)

Reply:
I see no problem removing it, but if we do i would recommend that we add the pop to the [_run_pipeline](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_internal/aad_client.py#L66) method so it does not make its way into the pipeline.

Yes. That's the reason we need to be careful to add new kwarg for get_token() method. :)

As you addressed, we need a more decent way to handle it. Currently, let's make `client_secret` invalid for get_token().

Reply:
So we want to raise an error on the kwarg of `client_secret` instead of quietly dropping it? I have not seen that in any other methods do you have an example somewhere? I assume it would just be a ValueError?

Reply:
We did not document `client_secret` as a valid parameter for `get_token()`.

For other arbitrary kwargs passed into get_token(), we raise errors.

Let's align with that.

Does this make sense?

Reply:
That last commit removes the `client_secret` from the `get_token` method. I do not see where they error code of checking kwargs anywhere but the error will be raised naturally by the `_run_pipeline` method due to that being an invalid argument. This will maintain the behavior that was seen previously.

Reply:
Thank you. The PR looks good to me.

Please fix the pipeline failures.

You can fix it by running 

python ./scripts/devops_tasks/validate_formatting.py --service_directory=identity --validate=True

under the <azure-sdk-for-python> folder.